File: deploy/upload_data.js
const fs = require('fs');
const path = require('path');
const http = require('http');

const data = JSON.parse(fs.readFileSync(path.join(__dirname, 'data_temp', 'jobs.json'), 'utf8'));
console.log(`数据大小: ${data.jobs.length} 条职位, ${data.companies.length} 家公司`);

const options = {
  hostname: '101.34.91.205',
  port: 3001,
  path: '/api/all-jobs',
  method: 'GET',
  timeout: 10000
};

const req = http.request(options, (res) => {
  let body = '';
  res.on('data', (chunk) => body += chunk);
  res.on('end', () => {
    try {
      const result = JSON.parse(body);
      console.log(`当前服务器数据: ${result.total} 条`);
    } catch (e) {
      console.log('获取当前数据失败:', e.message);
    }
    uploadData();
  });
});

req.on('error', (e) => {
  console.log('获取当前数据失败:', e.message);
  uploadData();
});

req.end();

function uploadData() {
  const postData = JSON.stringify(data);
  const options = {
    hostname: '101.34.91.205',
    port: 3001,
    path: '/api/upload-data',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(postData),
    },
    timeout: 60000
  };

  console.log('正在上传数据...');
  const req = http.request(options, (res) => {
    let body = '';
    res.on('data', (chunk) => body += chunk);
    res.on('end', () => {
      try {
        const result = JSON.parse(body);
        console.log('上传结果:', JSON.stringify(result).slice(0, 200));
      } catch (e) {
        console.log('上传响应:', body.slice(0, 200));
      }
      checkStatus();
    });
  });

  req.on('error', (e) => {
    console.log('上传失败:', e.message);
    checkStatus();
  });

  req.write(postData);
  req.end();
}

function checkStatus() {
  const options = {
    hostname: '101.34.91.205',
    port: 3001,
    path: '/api/status',
    method: 'GET',
    timeout: 10000
  };

  console.log('\n=== 检查上传后的数据 ===');
  const req = http.request(options, (res) => {
    let body = '';
    res.on('data', (chunk) => body += chunk);
    res.on('end', () => {
      try {
        const status = JSON.parse(body);
        console.log(`总职位数: ${status.totalJobs}`);
        console.log(`今日新增: ${status.todayNew}`);
        console.log(`最后更新: ${status.lastCrawl}`);
      } catch (e) {
        console.log('检查失败:', e.message);
      }
    });
  });

  req.on('error', (e) => {
    console.log('检查失败:', e.message);
  });

  req.end();
}

File: deploy/upload_data.py
import urllib.request
import urllib.parse
import json
import os

print('=== Uploading data to server ===')

data_path = r'C:\Users\44525\AppData\Roaming\TRAE SOLO CN\ModularData\ai-agent\work-mode-projects\6a5255c781d5bfa90b8e35c6\deploy\data_temp\jobs.json'

if not os.path.exists(data_path):
    print(f'数据文件不存在: {data_path}')
    exit(1)

with open(data_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

print(f'数据大小: {len(data["jobs"])} 条职位, {len(data["companies"])} 家公司')

url = 'http://101.34.91.205:3001/api/all-jobs'
try:
    with urllib.request.urlopen(url, timeout=10) as resp:
        result = json.loads(resp.read())
        print(f'当前服务器数据: {result["total"]} 条')
except Exception as e:
    print(f'获取当前数据失败: {e}')

url = 'http://101.34.91.205:3001/api/upload-data'
data_json = json.dumps(data, ensure_ascii=False).encode('utf-8')
req = urllib.request.Request(url, data=data_json, method='POST', headers={'Content-Type': 'application/json'})

try:
    print('正在上传数据...')
    with urllib.request.urlopen(req, timeout=60) as resp:
        result = json.loads(resp.read())
        print('上传结果:', result)
except urllib.error.HTTPError as e:
    print(f'HTTP错误: {e.code} - {e.read().decode("utf-8", errors="replace")}')
except Exception as e:
    print(f'上传失败: {e}')

print('\n=== 检查上传后的数据 ===')
url = 'http://101.34.91.205:3001/api/status'
try:
    with urllib.request.urlopen(url, timeout=10) as resp:
        status = json.loads(resp.read())
        print(f'总职位数: {status["totalJobs"]}')
        print(f'今日新增: {status["todayNew"]}')
        print(f'最后更新: {status["lastCrawl"]}')
except Exception as e:
    print(f'检查失败: {e}')

print('\n=== Done ===')